pricing/mutations.py

```diff
@@ -79,80 +75,100 @@ def mutate(root, info, **validated_data):
-        amount = graphene.Float(required=True)
-        amount_type = AmountTypeEnum(required=True)
-        active = graphene.Boolean(required=True)
-        num_sessions = graphene.Int(required=True)
+        discountId = graphene.ID()
+        name = graphene.String()
+        description = graphene.String()
+        amount = graphene.Float()
+        amount_type = AmountTypeEnum()
+        active = graphene.Boolean()
+        num_sessions = graphene.Int()
 
     multi_course_discount = graphene.Field(MultiCourseDiscountType)
+    created = graphene.Boolean()
 
     @staticmethod
     @staff_member_required
     def mutate(root, info, **validated_data):
-        multi_course_discount = MultiCourseDiscount.objects.create(**validated_data)
-        return CreateMultiCourseDiscount(multi_course_discount=multi_course_discount)
+        multi_course_discount, created = MultiCourseDiscount.objects.update_or_create(
+            id=validated_data.pop('discountId', None),
+            defaults=validated_data
+        )
+        return CreateMultiCourseDiscount(multi_course_discount=multi_course_discount, created=created)
 
 
 class CreateDateRangeDiscount(graphene.Mutation):
     class Arguments:
-        name = graphene.String(required=True)
-        description = graphene.String(required=True)
-        amount = graphene.Float(required=True)
-        amount_type = AmountTypeEnum(required=True)
-        active = graphene.Boolean(required=True)
-        start_date = graphene.types.datetime.Date(required=True)
-        end_date = graphene.types.datetime.Date(required=True)
+        discountId = graphene.ID()
+        name = graphene.String()
+        description = graphene.String()
+        amount = graphene.Float()
+        amount_type = AmountTypeEnum()
+        active = graphene.Boolean()
+        start_date = graphene.types.datetime.Date()
+        end_date = graphene.types.datetime.Date()
 
     date_range_discount = graphene.Field(DateRangeDiscountType)
+    created = graphene.Boolean()
 
     @staticmethod
     @staff_member_required
     def mutate(root, info, **validated_data):
-        date_range_discount = DateRangeDiscount.objects.create(**validated_data)
-        return CreateDateRangeDiscount(date_range_discount=date_range_discount)
+        date_range_discount, created = DateRangeDiscount.objects.update_or_create(
+            id=validated_data.pop('discountId', None),
+            defaults=validated_data
+        )
+        return CreateDateRangeDiscount(date_range_discount=date_range_discount, created=created)
 
 
 class CreatePaymentMethodDiscount(graphene.Mutation):
     class Arguments:
-        name = graphene.String(required=True)
-        description = graphene.String(required=True)
-        amount = graphene.Float(required=True)
-        amount_type = AmountTypeEnum(required=True)
-        active = graphene.Boolean(required=True)
-        payment_method =graphene.String(required=True)
-    
+        discountId = graphene.ID()
```

Comment:
leave python arguments as underscore - graphQL will automatically convert to camel case when you need to make a request